CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/fragments/CameraFragment.kt

```diff
@@ -275,62 +270,73 @@ class CameraFragment : Fragment() {
-                        Log.d(TAG, "Average luminosity: $luma. " +
-                                "Frames per second: ${"%.01f".format(fps)}")
-                    })
-        }
 
-        // Apply declared configs to CameraX using the same lifecycle owner
-        CameraX.bindToLifecycle(
-                viewLifecycleOwner, preview, imageCapture, imageAnalyzer)
+        val rotation = previewView.display.rotation
+
+        // Bind the cameraProvider to the LifeCycleOwner
+        val cameraSelector = CameraSelector.Builder().requireLensFacing(lensFacing).build()
+        val cameraProviderFuture = ProcessCameraProvider.getInstance(requireContext())
+        cameraProviderFuture.addListener(Runnable {
+
+            // CameraProvider
+            val cameraProvider: ProcessCameraProvider = cameraProviderFuture.get()
+
+            // Preview
+            preview = Preview.Builder()
+                .setTargetName("Preview")
+                .setTargetAspectRatio(screenAspectRatio)
+                .setTargetRotation(rotation)
+                .build()
+
+            // Default PreviewSurfaceProvider
+            preview?.setPreviewSurfaceProvider(previewView.previewSurfaceProvider)
+
+            // ImageCapture
+            capture = ImageCapture.Builder()
+                .setTargetName("Capture")
+                .setCaptureMode(ImageCapture.CAPTURE_MODE_MINIMIZE_LATENCY)
+                // We request aspect ratio but no resolution to match preview config but letting
+                // CameraX optimize for whatever specific resolution best fits requested capture mode
+                .setTargetAspectRatio(screenAspectRatio)
+                // Set initial target rotation, we will have to call this again if rotation changes
+                // during the lifecycle of this use case
+                .setTargetRotation(rotation)
+                .build()
+
+            // ImageAnalysis
+            analysis = ImageAnalysis.Builder()
+                .setTargetName("Analysis")
+                .setTargetAspectRatio(screenAspectRatio)
+                .setTargetRotation(rotation)
+                .build()
+
+            val analyzer = LuminosityAnalyzer {luma ->
+                // Values returned from our analyzer are passed to the attached listener
+                // We log image analysis results here - you should do something useful instead!
+                Log.d(TAG, "Average luminosity: $luma")
+            }
+            analysis?.setAnalyzer(mainExecutor, analyzer)
+
+            // Must unbind the use-cases before rebinding them.
+            cameraProvider.unbindAll()
+
+            try {
+                // A variable number of use-cases can be passed here.
+                val camera = cameraProvider.bindToLifecycle(
+                    this as LifecycleOwner, cameraSelector, preview, capture, analysis
+                )
+                cameraControl = camera.cameraControl
+                cameraInfo = camera.cameraInfo
```

Comment:
This is indeed a relic from trying to obtain values and controlling the camera. Need to see if it is used elsewhere, probably it can be removed altogether - however, for an example application it might make sense to provide the handles. Can change that.

Reply:
There still should be a comment and/or a `lateinit camera`, so that people will know how to get hold of it, which seems to cause some confusion.